numpy/lib/tests/test_polynomial.py

```diff
@@ -222,6 +222,20 @@ def test_poly_eq(self):
         assert_equal(p == p2, False)
         assert_equal(p != p2, True)
 
+    def test_poly_coeffs_immutable(self):
+        """ Coefficients should not be modifiable """
+        p = np.poly1d([1, 2, 3])
+
+        try:
+            # despite throwing an exception, this used to change state
+            p.coeffs += 1
+        except Exception:
+            pass
+        assert_equal(p.coeffs, [1, 2, 3])
+
+        p.coeffs[2] += 10
```

Comment:
The fact that this fails silently is possibly an argument for returning a readonly view instead.

However, that would cause surprise in cases like `c = p.coeffs; c[2] += 10`